common/src/main/java/org/opensearch/ml/common/transport/upload_chunk/MLUploadModelMetaInput.java

```diff
@@ -0,0 +1,230 @@
+        }
+        if (functionName == null) {
+            this.functionName = FunctionName.TEXT_EMBEDDING;
+        }
+        if (version == null) {
+            throw new IllegalArgumentException("model version is null");
+        }
+        if (modelFormat == null) {
+            throw new IllegalArgumentException("model format is null");
+        }
+        if (modelContentHash == null) {
+            throw new IllegalArgumentException("model content hash value is null");
+        }
+        if (modelConfig == null) {
+            throw new IllegalArgumentException("model config is null");
+        }
+        if (totalChunks == null) {
+            throw new IllegalArgumentException("total chunks field is null");
+        }
+        this.name = name;
+        this.functionName = functionName;
+        this.version = version;
+        this.description = description;
+        this.modelFormat = modelFormat;
+        this.modelState = modelState;
+        this.modelContentSizeInBytes = modelContentSizeInBytes;
+        this.modelContentHash = modelContentHash;
+        this.modelConfig = modelConfig;
+        this.totalChunks = totalChunks;
+    }
+
+    public MLUploadModelMetaInput(StreamInput in) throws IOException{
+        this.name = in.readString();
+        this.functionName = in.readEnum(FunctionName.class);
+        this.version = in.readString();
+        this.description = in.readOptionalString();
+        if (in.readBoolean()) {
+            modelFormat = in.readEnum(MLModelFormat.class);
+        }
+        if (in.readBoolean()) {
+            modelState = in.readEnum(MLModelState.class);
+        }
+        this.modelContentSizeInBytes = in.readOptionalLong();
+        this.modelContentHash = in.readString();
+        if (in.readBoolean()) {
+            modelConfig = new TextEmbeddingModelConfig(in);
+        }
+        this.totalChunks = in.readInt();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(name);
+        out.writeEnum(functionName);
+        out.writeString(version);
+        out.writeOptionalString(description);
+        if (modelFormat != null) {
+            out.writeBoolean(true);
+            out.writeEnum(modelFormat);
+        } else {
+            out.writeBoolean(false);
+        }
+        if (modelState != null) {
+            out.writeBoolean(true);
+            out.writeEnum(modelState);
+        } else {
+            out.writeBoolean(false);
+        }
+        out.writeLong(modelContentSizeInBytes);
+        out.writeString(modelContentHash);
+        if (modelConfig != null) {
+            out.writeBoolean(true);
+            modelConfig.writeTo(out);
+        } else {
+            out.writeBoolean(false);
+        }
+        out.writeInt(totalChunks);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        if (name != null) {
```

Comment:
Suggest remove the if null checking for mandatory fields as they never could be null. That means unit test can never cover the field ==  null case.